src/apps/projinfo.cpp

```diff
@@ -1248,6 +1263,76 @@ int main(int argc, char **argv) {
         }
     }
 
+    if (!listCRS.empty()) {
+        auto params = proj_get_crs_list_parameters_create();
+        bool all_types = false;
+        std::vector<PJ_TYPE> types;
+        std::vector<std::string> auth_names;
+        auto tokens = split(listCRS, ',');
+        for (auto token : tokens) {
+            auto auth_tokens = split(token, '=');
+            if (auth_tokens.size() == 2 && ci_equal(auth_tokens[0], "auth_name")) {
+                auth_names.push_back(auth_tokens[1]);
+                continue;
+            }
+
+            if (ci_equal(token, "allow_deprecated")) {
+                params->allow_deprecated = true;
+            } else if (ci_equal(token, "all_types")) {
+                all_types = true;
+            } else if (ci_equal(token, "geodetic")) {
+                types.push_back(PJ_TYPE_GEODETIC_CRS);
+            } else if (ci_equal(token, "geocentric")) {
+                types.push_back(PJ_TYPE_GEOCENTRIC_CRS);
+            } else if (ci_equal(token, "geographic")) {
+                types.push_back(PJ_TYPE_GEOGRAPHIC_CRS);
+            } else if (ci_equal(token, "geographic_2d")) {
+                types.push_back(PJ_TYPE_GEOGRAPHIC_2D_CRS);
+            } else if (ci_equal(token, "geographic_3d")) {
+                types.push_back(PJ_TYPE_GEOGRAPHIC_3D_CRS);
+            } else if (ci_equal(token, "vertical")) {
+                types.push_back(PJ_TYPE_VERTICAL_CRS);
+            } else if (ci_equal(token, "projected")) {
+                types.push_back(PJ_TYPE_PROJECTED_CRS);
+            } else if (ci_equal(token, "compound")) {
+                types.push_back(PJ_TYPE_COMPOUND_CRS);
+            } else if (ci_equal(token, "temporal")) {
+                types.push_back(PJ_TYPE_TEMPORAL_CRS);
+            } else if (ci_equal(token, "engineering")) {
+                types.push_back(PJ_TYPE_ENGINEERING_CRS);
+            } else if (ci_equal(token, "bound")) {
+                types.push_back(PJ_TYPE_BOUND_CRS);
+            } else if (ci_equal(token, "other")) {
+                types.push_back(PJ_TYPE_OTHER_CRS);
+            } else {
+                std::cerr << "Unrecognized value for option --list-crs: " << token
+                          << std::endl;
+                usage();
+            }
+        }
+
+        if (all_types) {
+            types.clear();
+        }
+        params->typesCount = types.size();
+        params->types = types.data();
+
```

Comment:
Could you take into account the --bbox and set it in the params ?
It would be good also to take into account --area and do post filtering

Reply:
Is there any way to extract the w,e,n,s values from the `Extent bboxFilter`?

Reply:
yes, but that's a bit convoluted. Simpler would be to just store them in dedicated local variables when parsing --bbox

Reply:
Ok. It is simple.
I tried `projinfo --list-crs vertical  --bbox 0,40,1,41` that included `Alicante height`. However `projinfo --list-crs vertical  --bbox 0,40,10,41` does not included it. It sounds strange to me. Is that right?

Reply:
see the crs_area_of_use_contains_bbox member of PROJ_CRS_LIST_PARAMETERS. What is missing in the doc is that its default value is TRUE

Reply:
I added `--bbox`. Including `--area` would be more complicated. It needs some code reorganization. Is it really useful?
In that case it would make sense to create an `Extent` (either from --bbox or --area), and intersect it on a post filter.
BTW, I set `crs_area_of_use_contains_bbox` to false.

Reply:
> Is it really useful?

It would be nice to have "--list-crs --area France" working. But that can be a later improvement. The .rst doc of --list-crs should mention which other options can be conbined with it

Reply:
>  But that can be a later improvement.

Yes.

>The .rst doc of --list-crs should mention which other options can be conbined with it

It already has as "Affected by options `--authority`, `--bbox`, `--main-db-path` and `--aux-db-path`". It that enough?

Reply:
> --area France

Do you mean to filter only the CRSs that exactly has the area name "France", or do a more complicated geometry, like area intersection?

Reply:
> Do you mean to filter only the CRSs that exactly has the area name "France", or do a more complicated geometry, like area intersection?

The idea would be to look that the name of the extent of the CRS contains the word France.

Reply:
I have another branch using C++ function directly, and including `--area`. I will do another PR (so you can compare) soon.